### PR TITLE
feat(planner): plan insert

### PR DESCRIPTION
**Pull request by skyzh**
As title

### CODE REVIEW

File: src/planner/plan_insert.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace bustub {
+
+auto Planner::PlanInsert(const InsertStatement &statement) -> std::unique_ptr<AbstractPlanNode> {
+  auto select = PlanSelect(*statement.select_);
+
+  auto table = catalog_.GetTable(statement.table_);
+  BUSTUB_ASSERT(table, "table not found");
+
+  return std::make_unique<InsertPlanNode>(SavePlanNode(std::move(select)), table->oid_);
```
Comment (lmwnshn) [inline src/planner/plan_insert.cpp:16]: Is the other Insert ctor (taking in a `vec<vec<values>> raw_values`) handled somewhere? This assumes that the insert statement is an INSERT INTO ... SELECT?

**Review by lmwnshn — APPROVED** [review]
Clarifying question, LGTM

File: src/planner/plan_insert.cpp (hunk 1 above)
Comment (skyzh) [reply src/planner/plan_insert.cpp:16]: It won't be used anymore after we integrated the new frontend. I'll remove it once we've merged all PRs.

Comment (skyzh) [reply src/planner/plan_insert.cpp:16]: Even for `INSERT INTO ... VALUES`, values will be planned as `ValuesExecutor`. Within `ValuesExecutor`, it stores `Vec<Vec<AbstractExpression>>`. We will never directly generate `Value` in the query processing layer.

**Pull request merged**